driver/src/main/com/mongodb/reactivestreams/client/MongoIterablePublisher.java

```diff
@@ -87,77 +82,145 @@ public void onResult(final AsyncBatchCursor<TResult> result, final Throwable t)
         @Override
         protected void handleCancel() {
             super.handleCancel();
-            AsyncBatchCursor<TResult> cursor = batchCursor.get();
-            if (cursor != null) {
-                cursor.close();
+            if (batchCursor != null) {
+                batchCursor.close();
             }
         }
 
-        void getNextBatch() {
-            log("getNextBatch");
-            if (batchCursorNextLock.compareAndSet(false, true)) {
-                final AsyncBatchCursor<TResult> cursor = batchCursor.get();
-                if (cursor.isClosed()) {
-                    cursorCompleted.set(true);
-                    batchCursorNextLock.set(false);
-                    processResultsQueue();
-                } else {
-                    int batchSize = wanted.get() > Integer.MAX_VALUE ? Integer.MAX_VALUE : wanted.intValue();
-                    cursor.setBatchSize(batchSize);
-                    cursor.next(new SingleResultCallback<List<TResult>>() {
-                        @Override
-                        public void onResult(final List<TResult> result, final Throwable t) {
-                            if (t != null) {
-                                onError(t);
-                                batchCursorNextLock.set(false);
-                            } else {
-                                if (result != null) {
-                                    resultsQueue.addAll(result);
-                                } else {
-                                    cursorCompleted.set(true);
-                                }
-                                batchCursorNextLock.set(false);
-                                processResultsQueue();
-                            }
-                        }
-                    });
+        private void processResultsQueue() {
+            log("processResultsQueue");
```

Comment:
At most these should be trace logging messages, not debug.  There is not enough context to make them very useful IMO.